vendor/engines/saas/lib/saas/engine.rb

```diff
@@ -17,6 +17,20 @@ class Engine < ::Rails::Engine
         })
       end
 
+      LoginController.class_eval do
+        before_action :page_job, only: [:public, :private]
+
+        :private
+
+        def page_job
+          auth_request = request.params['code'] && request.params['state']
+          if request.referer !~ /github\.com/ && !logged_in? && !auth_request
```

Comment:
Am I right in thinking the `request.referer` check is redundant given the other two?

1. `/login`
2. `/login/{scope}` (not `logged_in?` and `!auth_request`, so page is logged)
3. https://github.com/login/oauth/authorize
4. `/login/{scope}?code=...&state=...` (`auth_request` is true)
5. `/login/{scope}` (`logged_in?` should now be true)